c/authService.c

```diff
@@ -173,16 +196,223 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
     respondWithError(res, HTTP_STATUS_BAD_REQUEST, "Unexpected access level");
     return 0;
   }
-  /* printf("query: user %s, class %s, entity %s, access %d\n", userName, class,
-      entity, access); */
+
   privilegedServerName = getConfiguredProperty(service->server,
       HTTP_SERVER_PRIVILEGED_SERVER_PROPERTY);
   rc = zisCheckEntity(privilegedServerName, userName, class, entity, access,
       &reqStatus);
+
   respond(res, rc, &reqStatus);
   return 0;
 }
 
+int serveAuthCheckByParams(HttpService *service, char *userName, char *class, char *entity, int access, JsonObject *envSettings) {
+  int rc = 0;
+  int rbacParm;
+  Json *rbacObj = jsonObjectGetPropertyValue(envSettings, "ZWED_dataserviceAuthentication_rbac");
+  if (rbacObj == NULL) { // Env variable doesn't exist, so check server config
+    JsonObject *dataserviceAuth = jsonObjectGetObject(service->server->sharedServiceMem, "dataserviceAuthentication");
+    rbacParm = jsonObjectGetBoolean(dataserviceAuth, "rbac");
+  } else { // Use env variable value
+    rbacParm = jsonAsBoolean(rbacObj);
+  }
+  
+  CrossMemoryServerName *privilegedServerName = getConfiguredProperty(service->server,
+      HTTP_SERVER_PRIVILEGED_SERVER_PROPERTY);
+  ZISAuthServiceStatus reqStatus = {0};
+  if (!rbacParm) {
+    return rc; // When rbac isn't enabled, we don't try to check the auth query
+  }
+  rc = zisCheckEntity(privilegedServerName, userName, class, entity, access,
+      &reqStatus);
+  zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RBAC check occurred for entity '%s' class '%s' access '%d' , rc: %d", entity, class, access, rc);
+  return rc;
+}
+
+int getProfileNameFromRequest(char *profileName, StringList *parsedFile, char *method, int instanceID, HttpResponse *response) {
```

Comment:
extractQuery seems to just loop through the query checking if StringListElt->next exists and setting entity and accessStr (because the query is already a SAF query) to use zisCheckEntity

getProfileNameFromRequest generates a query which we then use zisCheckEntity on. So extractQuery doesn't accomplish much because we already get entity and accessStr as part of the process of turning the URL into a SAF query